### PR TITLE
Update pallets to get teeracle with multiple sources and multiple trading pairs 

### PR DESCRIPTION
**Pull request by echevrier**
See pallets issue: https://github.com/integritee-network/pallets/issues/25

### CODE REVIEW

**Comment by brenzi** [comment]
`--features runtime-benchmarks` won't build

**Comment by brenzi** [comment]
benchmarking is ok for now. but running benchmarks on drop probably depends on too many factors

**Review by brenzi — APPROVED** [review]

File: runtime/src/weights/pallet_teeracle.rs [hunk 1]
```diff
@@ -18,35 +18,35 @@
...
 use frame_support::{traits::Get, weights::Weight};
 use sp_std::marker::PhantomData;
 
-/// Weight functions for pallet_teeracle.
+/// Weight functions for `pallet_teeracle`.
 pub struct WeightInfo<T>(PhantomData<T>);
 impl<T: frame_system::Config> pallet_teeracle::WeightInfo for WeightInfo<T> {
-	// Storage: Teerex Whitelist (r:1 w:1)
+	// Storage: Teerex EnclaveIndex (r:1 w:0)
+	// Storage: Teerex EnclaveRegistry (r:1 w:0)
```
Comment (echevrier) [inline runtime/src/weights/pallet_teeracle.rs:32]: Update exchange rate doesn't read the Teerex EnclaveRegistry. It only check if the enclave is in the EnclaveIndex

Comment (brenzi) [reply runtime/src/weights/pallet_teeracle.rs:32]: we have a somewhat reasonable value now. will accept it as is

**Pull request merged**